uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/CiviHRStatsGatherer.php

```diff
@@ -0,0 +1,172 @@
+  }
+
+  /**
+   * Fetch and set all required statistics.
+   *
+   * @return CiviHRStatistics
+   */
+  public function gather() {
+    $stats = new CiviHRStatistics();
+    $stats->setGenerationDate(new \DateTime());
+    $stats->setSiteName($this->cmsVariableService->get('site_name'));
+    $this->setBaseUrl($stats);
+    $this->setEntityCounts($stats);
+    $this->setContactSubtypes($stats);
+    $this->setReportConfigurations($stats);
+    $this->setAgeGroups($stats);
+
+    return $stats;
+  }
+
+  /**
+   * Sets the site base URL
+   *
+   * @param CiviHRStatistics $stats
+   */
+  private function setBaseUrl(CiviHRStatistics $stats) {
+    $config = & CRM_Core_Config::singleton();
+    $stats->setSiteUrl($config->userFrameworkBaseURL);
+  }
+
+  /**
+   * Fetches counts and sets them for all required entities
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setEntityCounts(CiviHRStatistics $stats) {
+    $taskAssignmentsKey = 'uk.co.compucorp.civicrm.tasksassignments';
+    $leaveAndAbsenceKey = 'uk.co.compucorp.civicrm.hrleaveandabsences';
+    $recruitmentKey = 'org.civicrm.hrrecruitment';
+
+    $stats->setEntityCount('contact', $this->getEntityCount('Contact'));
+
+    if (ExtensionHelper::isExtensionEnabled($taskAssignmentsKey)) {
+      $stats->setEntityCount('assignment', $this->getEntityCount('Assignment'));
+      $stats->setEntityCount('task', $this->getEntityCount('Task'));
+      $stats->setEntityCount('document', $this->getEntityCount('Document'));
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($leaveAndAbsenceKey)) {
+      // leave requests in last 100 days
+      $format = 'Y-m-d H:i:s';
+      $oneHundredDaysAgo = (new \DateTime('today - 100 days'))->format($format);
+      $params = ['from_date' => ['>=' => $oneHundredDaysAgo]];
+      $last100DaysCount = (int) civicrm_api3('LeaveRequest', 'getcount', $params);
+      $stats->setEntityCount('leaveRequestInLast100Days', $last100DaysCount);
+
+      // total leave requests
+      $leaveRequestCount = $this->getEntityCount('LeaveRequest');
+      $stats->setEntityCount('leaveRequest', $leaveRequestCount);
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($recruitmentKey)) {
+      $stats->setEntityCount('vacancy', $this->getEntityCount('HRVacancy'));
+    }
+
+    $stats->setEntityCount('drupalUser', $this->getEntityCount('UFMatch'));
+  }
+
+  /**
+   * Gets the number of a certain entity in the system.
+   *
+   * @param string $entity
+   *
+   * @return int
+   */
+  private function getEntityCount($entity) {
+    return (int) civicrm_api3($entity, 'getcount');
+  }
+
+  /**
+   * Fetches contact subtypes
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setContactSubtypes(CiviHRStatistics $stats) {
+    $params = ['parent_id' => ['IS NULL' => 1]];
+    $contactTypes = civicrm_api3('ContactType', 'get', $params)['values'];
```

Comment:
I'm not sure if including the count for disabled subtypes is useful either.

Disabling the subtype doesn't affect those contacts in that subtype. If a client really doesn't want to use that subtype then they can delete it, in which case it will not be included in the stats. 

In terms of how this might look on a graph in the future it could look like the number for contact subtypes jumped from a high number to zero and then back to a high number again if it was disabled and re-enabled, so I'd probably go with including them.

Reply:
> If a client really doesn't want to use that subtype then they can delete it, in which case it will not be included in the stats.

Or they can disable it. It's really up to them to decide.

Anyway, I think it's more up to Jamie/Mike/Guanhuan to decide if inactive types should be included or not

Reply:
I left a comment for Mike about this too.